### ISSUE TITLE
Formatting analyzer is extremely noisy for object initializers.

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#57279]
```
class Test
{
    public int P { get; set; }
    public Test(int x, int y) { }

    void M()
    {
        _ = new Test(1, 2) { P = 1 };
        _ = new Test(3,
                     4) { P = 1 };
        _ = new Test(5,
                     6){ P = 1 };
        _ = new Test(5,
                     6)
            { P = 1 };
    }
}
```

Observed: Formatting analyzer complains in three places, see
![image](https://user-images.githubusercontent.com/10522107/138169984-a2b37578-0c92-4284-90a0-dc93c88564e8.png)

Expected: A warning only in one place, the case of a missing space between ```)``` and ```{```.

**Comment by AlekseyTs** [comment]
> I believe all of the warnings above adhere to the current formatting design. 

Perhaps that is the case. However, I don't believe the current design works well, especially when object initializers are used inside more complicated expressions. The requirement to align braces at the beginning of the line feels completely unreasonable. The recommended formatting produces code that is extremely difficult to read. I would never format code like that. I guess the same is applicable to formatting of other expressions that involve braces.

**Comment by sharwell** [comment]
Can you provide more detail regarding the exact change request here? We'll have to take it to a design review with the understanding that any change to formatter output that doesn't preserve formatting for current users is a breaking change for anyone that runs FormattingAnalyzer in CI.
